--- a/net/unix/af_unix.c
+++ b/net/unix/af_unix.c
@@ -2877,6 +2877,15 @@ static int unix_seq_show(struct seq_file *seq, void *v)
 	else {
 		struct sock *s = v;
 		struct unix_sock *u = unix_sk(s);
+#ifdef CONFIG_KSU_SUSFS
+		if (current_uid().val >= 10000 && u->addr && UNIX_ABSTRACT(s)) {
+			int path_len = u->addr->len - sizeof(short);
+			if (path_len >= 32) {
+				return 0;
+			}
+		}
+#endif
 		unix_state_lock(s);
 
 		seq_printf(seq, "%pK: %08X %08X %08X %04X %02X %5lu",
