--- a/fs/proc/cmdline.c
+++ b/fs/proc/cmdline.c
@@ -4,8 +4,31 @@
 #include <linux/init.h>
 #include <linux/proc_fs.h>
 #include <linux/seq_file.h>
+#ifdef CONFIG_KSU_SUSFS
+#include <linux/cred.h>
+#include <linux/string.h>
+#include <linux/slab.h>
+#endif
 
 static int cmdline_proc_show(struct seq_file *m, void *v)
 {
+#ifdef CONFIG_KSU_SUSFS
+	if (current_uid().val >= 10000) {
+		char *buf = kstrdup(saved_command_line, GFP_KERNEL);
+		if (buf) {
+			char *p;
+			if ((p = strstr(buf, "androidboot.verifiedbootstate=orange")))
+				memcpy(p + 30, "green ", 6);
+			if ((p = strstr(buf, "androidboot.flash.locked=0")))
+				p[25] = '1';
+			if ((p = strstr(buf, "androidboot.vbmeta.device_state=unlocked")))
+				memcpy(p + 32, "locked  ", 8);
+			seq_puts(m, buf);
+			kfree(buf);
+			seq_putc(m, '\n');
+			return 0;
+		}
+	}
+#endif
 	seq_puts(m, saved_command_line);
 	seq_putc(m, '\n');
 	return 0;
